### ISSUE TITLE
WFS Pre-defined: validation of StoredQuery fails on a recommendation

### ISSUE THREAD

**Issue by thijsbrentjens** [INSPIRE-MIF/helpdesk-validator#42]
See WFS https://geodata.nationaalgeoregister.nl/cbsprovincies/wfs?&request=DescribeStoredQueries&service=WFS&version=2.0.0
Run the test for WFS Pre-defined datasets.

The test fails on the check "Contains the recommended 'GetSpatialDataset' identifier":

```
The recommended identifier 'http://inspire.ec.europa.eu/operation/download/GetSpatialDataSet', which should be used for serving pre-defined Spatial Data Sets, was not found.
```
This identifier is indeed not available, but this is a recommendation (TG Recommendation 13), not a TG requirement. The validation should not fail on this.

**Comment by thijsbrentjens** [comment]
Note that a check on this identifier is not part of the ATS either: https://github.com/inspire-eu-validation/download-service/blob/master/wfs-pre-defined/predefined-stored-query.md

**Comment by inakidiazdecerio** [comment]
Dear @thijsbrentjens,

Thank you for using the validator and for the message. 

There are two possible solutions for this issue, relax the ETS eliminating the validation or modify the ATS to add that validation.

We are going to discuss the possible solution next subgroup meeting.

Regards.

**Comment by michellutz** [comment]
I agree that we should revisit the requirements and recommendations in sections 6.4 and 6.5 of the WFS/Atom TG. Note that a change in the ATS would of course also mean a change in the TGs.

**Comment by thijsbrentjens** [comment]
Thanks for the replies. I'd prefer not to change the ATS or TG. I think the validator should not check a recommendation as if it were a requirement. This could confuse a user of the validator. Or there should be something like a warning, but this is not possible at the moment AFAIK.

**Comment by michellutz** [comment]
Ok, I agree. Looking again at TG requirements 49-51, I think the test should check whether there is at least one stored query that allows any "possible (i.e. available) combinations of CRS/DataSetIdCode/DataSetIdNamespace/language" to be queried, i.e. a stored query that has all these query parameters.

Now if there is just one CRS, DataSet or language, these parameters strictly speaking would not have to be offered, so we should also not check for all the parameters to be present. Or we would need to make this a manual check in case there is no stored query with the recommended id or with all the required parameters.

Also, we should probably check whether there is only one such stored query (with the required parameters) if there is none with the recommended name, so that a client knows which one is the "INSPIRE predefined data set" query.

In conclusion, I think we will need to rephrase this section in the TG (and ATS) to be more clear.

**Comment by MarcoMinghini** [comment]
**2017.4 meeting 2019-05-22**

It is agreed to relax the test as described above, i.e.:

if there is no stored query with the recommended id, or if there is no stored query with all the required parameters, make it a manual check (check that there is only one possible combination of the required parameter(s) involved).

Later, we can have a discussion with the MIG-T on whether the TG should be also rephrased.

**Comment by josemasensio** [comment]
Dear @thijsbrentjens,

We are happy to announce that the solution for this issue has been implemented in the INSPIRE Validator. The solution passed our quality controls, but we will appreciate if you could test it and give us some feedback. You can test the solution in the staging environment:

http://staging-inspire-validator.eu-west-1.elasticbeanstalk.com/etf-webapp/

Thanks in advance for your help.

**Comment by thijsbrentjens** [comment]
Thanks, I re-tested this too, it is also fixed in the staging environment.
